src/Interpreters/AsynchronousInsertQueue.cpp

```diff
@@ -184,7 +184,8 @@ void AsynchronousInsertQueue::push(ASTPtr query, ContextPtr query_context)
     if (!FormatFactory::instance().isInputFormat(insert_query.format))
         throw Exception(ErrorCodes::UNKNOWN_FORMAT, "Unknown input format {}", insert_query.format);
 
-    query_context->checkAccess(AccessType::INSERT, insert_query.table_id, sample_block.getNames());
+    if (insert_query.table_id)
```

Comment:
And if it's a table function, how do we check the access?

Reply:
[getTable](https://github.com/ClickHouse/ClickHouse/blob/ba9150abb00b31234ec1a6b426f3b36978f068c8/src/Interpreters/AsynchronousInsertQueue.cpp#L181) -> [execute](https://github.com/ClickHouse/ClickHouse/blob/ba9150abb00b31234ec1a6b426f3b36978f068c8/src/Interpreters/InterpreterInsertQuery.cpp#L63) -> [checkAccess](https://github.com/ClickHouse/ClickHouse/blob/ba9150abb00b31234ec1a6b426f3b36978f068c8/src/TableFunctions/ITableFunction.cpp#L21)

Reply:
Let's add a comment.